### ISSUE TITLE
train No.3

### ISSUE THREAD

**Issue by RCpengnan** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#154]
Hello! I want to ask two questions. Firstly, I tried to train the No.3 (--from-mobilenet)and train 285000 iters. However, I did not detect any points and lines when I ran the Demo with it. Do you know what the problem is?Is it because I didn't train for step four or five? Secondly, Can you tell me how much you lost at the end of your training? I want to know the approximate loss when we can stop training. I trained 285000 iters, my losses are as follows:
![image](https://user-images.githubusercontent.com/74439931/111891134-66ed8100-8a2b-11eb-8983-b508ebb94687.png)

**Comment by RCpengnan** [comment]
I ran into this problem during training, and I commented out this line of code in train.py(#evaluate(val_labels, val_output_name, val_images_folder, net)).Is this line of code commented out, so there is no result? Do you have any solutions?
![image](https://user-images.githubusercontent.com/74439931/111891402-e5e3b900-8a2d-11eb-90ed-1e3fdfd5fa5a.png)

**Comment by RCpengnan** [comment]
Hello,I have seen your previous reply. Now I am making 19 key points and have modified the code. The data sets I used are from Val2017 and Train2017 downloaded from the official website of Coco. Do you mean that I need to re-label the data? Isn't the data set downloaded from the official website of Coco already marked?
![image](https://user-images.githubusercontent.com/74439931/111894528-35ce7a00-8a46-11eb-8d80-2840bfb6fcc4.png)

**Comment by Daniil-Osokin** [comment]
Hi! First of all check if provided pre-trained model works for validation. This will say if validation works. Then check [output file](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/val.py#L157) with prediction results from your checkpoint. If it is empty, then such error may occur. So, if it is empty, try to visualize the [avg_heatmaps](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/val.py#L127) to see if anything was detected. Loss curves are here: #10, yours look reasonable.

**Comment by RCpengnan** [comment]
> 你好！首先，检查提供的预训练模型是否可用于验证。这将说明验证是否有效。然后，使用检查点的预测结果检查[输出文件](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/val.py#L157)。如果为空，则可能会发生此错误。因此，如果为空，请尝试可视化[avg_heatmaps](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/val.py#L127)以查看是否检测到任何东西。损耗曲线在这里：[＃10](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/issues/10)，您看起来很合理。

Thank you for your prompt reply. Thank you for your advice. I'll have a try. I also want to ask you that the key point I want to set now is 20, and I want to add a waist node. I would like to ask you if I can directly use the COCO data set. However, the COCO data set only annotates 18 key points. If I want to set 20 key points, do I need to annotate the COCO data set again?

**Comment by RCpengnan** [comment]
> Hi! First of all check if provided pre-trained model works for validation. This will say if validation works. Then check [output file](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/val.py#L157) with prediction results from your checkpoint. If it is empty, then such error may occur. So, if it is empty, try to visualize the [avg_heatmaps](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/val.py#L127) to see if anything was detected. Loss curves are here: #10, yours look reasonable.

Thank you for your prompt reply. Thank you for your advice. I'll have a try. I also want to ask you that the key points I want to set now is 20, and I want to add a waist node. I would like to ask you if I can directly use the COCO data set?  However, the COCO data set only annotates 18 key points. If I want to set 20 key points, do I need to annotate the COCO data set again?

**Comment by RCpengnan** [comment]
> Hi! First of all check if provided pre-trained model works for validation. This will say if validation works. Then check [output file](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/val.py#L157) with prediction results from your checkpoint. If it is empty, then such error may occur. So, if it is empty, try to visualize the [avg_heatmaps](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/val.py#L127) to see if anything was detected. Loss curves are here: #10, yours look reasonable.

Thank you for your prompt reply. Thank you for your advice. I'll have a try. I also want to ask you two question. Firstly, the key points I want to set now is 20, and I want to add a waist node. I would like to ask you if I can directly use the COCO data set? However, the COCO data set only annotates 18 key points. If I want to set 20 key points, do I need to annotate the COCO data set again?
Secondly,the key points can be detected if only the third step is carried out,?or the key points and connections can be detected only after the complete training steps have been carried out?
Looking forward to your reply! Thank you!

**Comment by gmakoudi** [comment]
Τα losses του osokin
![55278070-af7c7300-5318-11e9-9b92-4aebc0cc9b46](https://user-images.githubusercontent.com/24383032/112029351-6372fc00-8af6-11eb-9550-b8b1c75eb2df.png)

**Comment by Daniil-Osokin** [comment]
Yes, you should label somehow waist for the persons, either manually or compute from existing ones. Keypoints can be detected after the third step (you see, there is validation step during training, so after some initial iterations, e.g. 5000, keypoints are detected, try to visualize the heatmaps).

**Comment by RCpengnan** [comment]
> Yes, you should label somehow waist for the persons, either manually or compute from existing ones. Keypoints can be detected after the third step (you see, there is validation step during training, so after some initial iterations, e.g. 5000, keypoints are detected, try to visualize the heatmaps).
Thank you for your prompt reply. I get it!
